doc/groups.md

```diff
@@ -1 +1,573 @@
+200 | OK | OK
+```
+{
+    "status_code": "CDC-500",
+    "status_message": "Internal Server Error",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": [
+      {
+        "id": 1,
+        "title": "Admin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00"
+      },
+      {
+        "id": 2,
+        "title": "Sales",
+        "description": "Sales Admin",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00"
+      }
+} 
+```
+## <a name="view-detail"></a>View Detail Groups by id
+
+### Endpoint
+GET /groups/{id}
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payload
+
+No Request Payload
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---  
+404 | Not Found | Groups not found in database  
+500 | Internal Server Error | some un-handle error in server 
+200 | OK | OK
+
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "Data not found",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data":
+      {
+        "id": 2,
+        "title": "Sales",
+        "description": "Sales Admin",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00",
+        "aros_id": 2
+      }
+} 
+```
+
+### Logic
+
+#### Endpoint Validation
+- must be add parameter id
+- id must exist in database
+
+#### Headers Validation
+No Headers Validation
+
+#### Request Payloads Validation
+No Request Payloads Validation
+
+### Scenario Test
+
+#### Case : Negative Case 1
+- id not exist in database
+
+Endpoint : /groups/100
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "data not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case
+
+- id is exist in database
+
+Endpoint : /groups/2
+
+Headers :
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data":
+      {
+        "id": 2,
+        "title": "Sales",
+        "description": "Sales Admin",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00",
+        "aros_id": 2
+      }
+} 
+```
+
+## <a name="delete"></a>Delete Groups
+
+### Endpoint
+DELETE /groups/{id}
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | *
+Accept | application/json
+
+### Request Payloads
+No request payloads
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+404 | Not Found | Groups not found in database  
+500 | Internal Server Error | some un-handle error in server 
+204 | No Content | Server request has succeeded but response is no content
+
+Status 204 : No Response
+
+### Logic
```

Comment:
add logic : after delete group also delete aros where model = Groups and foreign_key = id of group